### PR TITLE
add flatbuffers version 23.3.3

### PR DESCRIPTION
**Pull request by CedricSchmeits**
Specify library name and version:  **flatbuffers/23.3.3**

adding v23.3.3 to the flatbuffers package

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by CedricSchmeits** [comment]
> I detected other pull requests that are modifying flatbuffers/all recipe:
> 
> * [flatbuffers: modernize more for conan v2 #16269](https://github.com/conan-io/conan-center-index/pull/16269)
> 
> This message is automatically generated by https://github.com/ericLemanissier/conan-center-conflicting-prs so don't hesitate to report issues/improvements there.

This pr #16269 does not add the v23.3.3

File: recipes/flatbuffers/all/conanfile.py [hunk 1]
```diff
@@ -128,7 +128,11 @@ def build(self):
         cmake.build()
 
     def package(self):
-        copy(self, "LICENSE.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        license_file_name = "LICENSE"
+        if Version(self.version) < "23.3.3":
+            license_file_name += ".txt"
+
+        copy(self, license_file_name, src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```
Comment (SpaceIm) [inline recipes/flatbuffers/all/conanfile.py:135]:
```suggestion
        copy(self, "LICENSE*", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```

Keep it simple

**Comment by CedricSchmeits** [comment]
@SpaceIm Do you know what is going wrong with the `c3i/conan-v2/pr-merge` step that has failed? When I go to that link I'm getting a: 403 Forbidden

**Comment by SpaceIm** [comment]
v2 pipeline can't pass for flatbuffers without https://github.com/conan-io/conan-center-index/pull/16269

**Comment by CedricSchmeits** [comment]
Well maybe the actual question is: Can this pull request be completed before that #16269 is completed? Or should I just wait with this?

**Review by danimtb — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**